File: 프로그래머스/2/42587. 프로세스/프로세스.js
function solution(priorities, location) {
    let answer = 0;
    let arr = []
    let max_value = Math.max(...priorities)
    
    for (let i = 0; i < priorities.length; i++) {
        arr.push(i)
    }
    
    while (priorities.length != 0) {
         max_value = Math.max(...priorities)
        
        if (priorities[0] < max_value) {
            priorities.push(priorities.shift())
            arr.push(arr.shift())
        } else {
            answer += 1
            priorities.shift()
            if (arr.shift() == location)
                return answer
        }
    }
}

File: 프로그래머스/2/42587. 프로세스/프로세스.py
def solution(priorities, location):
    answer = 0
    arr = list(range(len(priorities)))
    
    while len(priorities) != 0:
        max_value = max(priorities)
        
        if priorities[0] < max_value:
            priorities.append(priorities.pop(0))
            arr.append(arr.pop(0))
        else:
            answer += 1
            priorities.pop(0)
            if arr.pop(0) == location:
                return answer